ios/Classes/SwiftAppodealFlutterPlugin.swift

```diff
@@ -81,16 +82,22 @@ public class SwiftAppodealFlutterPlugin: NSObject, FlutterPlugin {
         result(Appodeal.isReadyForShow(with: adType))
     }
 
+    private func canShow(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
+        let args = call.arguments as! Dictionary<String, Any>
+        let adType = getAdType(adId: args["adType"] as! Int)
+        let placementName = args["placementName"] as? String ?? ""
+        result(Appodeal.canShow(adType, forPlacement: placementName))
+    }
+
     private func show(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
         let args = call.arguments as! [String: Any]
         let adType = getShowStyle(adType: getAdType(adId: args["adType"] as! Int))
-
+        let placementName = args["placementName"] as? String ?? ""
```

Comment:
As I can see here, when the user doesn't specify a placement name for the ad, it defaults the placement name to an empty string, but I would rather use the same approach that was used for Android when there's no placement name. In other words, check if the `placementName` is nil and then call the appropriate function.

Something like this:


```swift
if let placementName = args["placementName"] as? String {
    result(Appodeal.showAd(adType, forPlacement: placementName, rootViewController: rootViewController))
} else {
    result(Appodeal.showAd(adType, rootViewController: rootViewController))
}
```

My reason for this is because when we call `Appodeal.showAd` without the `forPlacement` parameter, the SDK will use the default placement defined in your Appodeal account. However, if the user doesn't define a placement name and we try to call `Appodeal.showAd` with an empty placement (`""`) like in the existing code, then the `Appodeal.showAd` will not try to use the default placement in the Appodeal account; it will actually try to find a placement with an empty name.

What do you think?

Reply:
Nice catch, yes much better to only send the placement name if we have one 👍